shared/actions/fs/platform-specific.desktop.js

```diff
@@ -288,10 +288,31 @@ function installCachedDokan() {
   })
 }
 
+function uninstallDokan(uninstallString: string): Promise<boolean> {
+  return new Promise((resolve, reject) => {
+    logger.info('Invoking dokan uninstaller')
+    try {
+      execSync(uninstallString, {windowsHide: true})
+      resolve(true)
+    } catch (err) {
+      logger.error('uninstallDokan caught', err)
+      reject(err)
+    }
+  })
+}
+
 function installDokanSaga() {
   return Saga.call(installCachedDokan)
 }
 
+function* uninstallDokanSaga() {
+  const state: TypedState = yield Saga.select()
+  const uninstallString = Constants.kbfsUninstallString(state)
+  if (uninstallString) {
+    yield Saga.call(uninstallDokan, uninstallString)
+  }
+}
```

Comment:
If you refactor this a bit, you can use `actionToPromise` below.
```javascript
const uninstallDokanPromise = (state: TypedState) => {
  const uninstallString = Constants.kbfsUninstallString(state)
  if (uninstallString) {
    return uninstallDokan(uninstallString)
  }
  return new Promise(resolve => resolve())
}
```

Reply:
good idea, thanks

Reply:
actually, I found I wanted to get the fuse status again after the synchronous uninstall, so I changed it back except not pure.